pkg/plugin.go

```diff
@@ -79,25 +79,32 @@ func newDatasource() datasource.ServeOpts {
 }
 
 // QueryData handles multiple queries and returns multiple responses.
-// req contains the queries []DataQuery (where each query contains RefID as a unique identifer).
+// req contains the queries []DataQuery (where each query contains RefID as a unique identifier).
 // The QueryDataResponse contains a map of RefID to the response for each query, and each response
 // contains Frames ([]*Frame).
 func (td *SampleDatasource) QueryData(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
-	// create response struct
 	response := backend.NewQueryDataResponse()
+	results := make(chan func(backend.Responses), len(req.Queries))
 
-	// loop over queries and execute them individually.
+	// Execute the queries in parallel and collect the responses.
```

Comment:
Is there a limit on the length of `req.Queries`? BQ has a limit of 100 concurrent queries. Unless `req.Queries` is limited somewhere else, you might want to impose a concurrency limit here.